airflow/plugins/operators/scrape_ntd_xlsx.py

```diff
@@ -120,7 +121,15 @@ def execute(self, context, *args, **kwargs):
         download_url = self.raw_excel_extract.file_url
 
         if self.product == "complete_monthly_ridership_with_adjustments_and_estimates":
-            download_url = pull_url_from_xcom(context=context)
+            download_url = pull_url_from_xcom(key="ridership_url", context=context)
+
+        if self.product == "annual_database_agency_information":
+            download_url = pull_url_from_xcom(key="agency_url", context=context)
+
+        if self.product == "annual_database_contractual_relationship":
+            download_url = pull_url_from_xcom(
+                key="contractual_relationship_url", context=context
+            )
 
         # see what is returned
         logging.info(f"reading ridership url as {download_url}")
```

Comment:
Maybe needs to be replaced by the product name instead of "ridership".

Reply:
great catch! just revised